pkg/apis/keycloak/v1alpha1/keycloakrealm_types.go

```diff
@@ -190,3 +198,19 @@ type KeycloakRealmList struct {
 func init() {
 	SchemeBuilder.Register(&KeycloakRealm{}, &KeycloakRealmList{})
 }
+
+func (i *KeycloakRealm) UpdateStatusSecondaryResources(kind string, resourceName string) {
```

Comment:
This is the third time I see this kind of structure (one for `Keycloak` CR, one here and one in my backup PR). Perhaps we could implement it as a utility function somewhere?

Reply:
Completely agree. Will split this out this morning

Reply:
New commit up with this split out. I am not happy with the implementation though and would love your feedback on it

Reply:
Looks good!